### ISSUE TITLE
vm create-from: IOType is implicitly determined from the MediaLink.

### ISSUE THREAD

**Issue by janihur** [Azure/azure-xplat-cli#1514]
When I `vm export` and `vm create-from` a Linux virtual machine I end with an error:

```
IOType is implicitly determined from the MediaLink. This should not be explicitly specified in the request.
```
I have to remove the following entry from the role file before the `create-from` command succeeds:

```
"iOType":"Standard"
```

A full example:

```
$ azure vm export devoops-vm-1 devoops-vm-1.json
info:    Executing command vm export
+ Getting virtual machines                                                     
+ Exporting the VM                                                             
info:    VM devoops-vm-1 exported to devoops-vm-1.json
info:    vm export command OK
$ cat devoops-vm-1.json 
{"configurationSets":[{"inputEndpoints":[{"localPort":22,"name":"SSH","port":323,"protocol":"tcp","virtualIPAddress":"xxx.xxx.xxx.xxx","enableDirectServerReturn":false,"endpointAcl":{"rules":[{"order":0,"action":"permit","remoteSubnet":"xxx.xxx.xxx.xxx/32","description":"home office"}]}}],"networkInterfaces":[],"publicIPs":[],"storedCertificateSettings":[],"subnetNames":[],"configurationSetType":"NetworkConfiguration"}],"dataVirtualHardDisks":[],"resourceExtensionReferences":[],"roleName":"devoops-vm-1","oSVersion":"","roleType":"PersistentVMRole","oSVirtualHardDisk":{"hostCaching":"ReadWrite","name":"devoops-vm-1-devoops-vm-1-0-201501031828280231","operatingSystem":"Linux","iOType":"Standard"},"roleSize":"Basic_A0","provisionGuestAgent":true}
$ azure vm delete devoops-vm-1
info:    Executing command vm delete
+ Getting virtual machines                                                     
Delete the VM devoops-vm-1 ? [y/n] y
+ Deleting VM                                                                  
+ Deleting Cloud Service                                                       
info:    vm delete command OK
$ azure vm create-from --verbose --location "West Europe" --ssh-cert devoops-vm.pem devoops-vm-1 devoops-vm-1.json 
info:    Executing command vm create-from
verbose: Loading role file: devoops-vm-1.json
verbose: Trying to open SSH cert: devoops-vm.pem
verbose: Checking for existence of devoops-vm-1 cloud service
verbose: Looking up cloud service
verbose: Registering sqlserver with function registerProvider
verbose: Registering resource sqlserver with subscription <SUBSCRIPTION_ID>
info:    cloud service devoops-vm-1 not found.
verbose: Resolving the location West Europe
verbose: Registering sqlserver with function registerProvider
verbose: Registering resource sqlserver with subscription <SUBSCRIPTION_ID>
verbose: Location resolved to West Europe
verbose: Creating cloud service
verbose: Registering sqlserver with function registerProvider
verbose: Registering resource sqlserver with subscription <SUBSCRIPTION_ID>
verbose: Configuring certificate
verbose: uploading cert
verbose: Registering sqlserver with function registerProvider
verbose: Registering resource sqlserver with subscription <SUBSCRIPTION_ID>
verbose: Registering sqlserver with function registerProvider
verbose: Registering resource sqlserver with subscription <SUBSCRIPTION_ID>
verbose: uploading cert succeeded
verbose: Creating VM
verbose: Registering sqlserver with function registerProvider
verbose: Registering resource sqlserver with subscription <SUBSCRIPTION_ID>
verbose: Error occurred. Deleting devoops-vm-1 cloud service
verbose: Deleting cloud service
verbose: Registering sqlserver with function registerProvider
verbose: Registering resource sqlserver with subscription <SUBSCRIPTION_ID>
verbose: Cloud service devoops-vm-1 deleted
error:   IOType is implicitly determined from the MediaLink. This should not be explicitly specified in the request.
info:    Error information has been recorded to azure.err
verbose: Error: IOType is implicitly determined from the MediaLink. This should not be explicitly specified in the request.
    at Function.ServiceClient._normalizeError (/usr/local/lib/node_modules/azure-cli/node_modules/azure-common/lib/services/serviceclient.js:809:23)
    at /usr/local/lib/node_modules/azure-cli/node_modules/azure-common/lib/services/filters/errorhandlingfilter.js:44:29
    at Request._callback (/usr/local/lib/node_modules/azure-cli/node_modules/azure-common/lib/http/request-pipeline.js:109:14)
    at Request.self.callback (/usr/local/lib/node_modules/azure-cli/node_modules/request/request.js:129:22)
    at Request.EventEmitter.emit (events.js:98:17)
    at Request.<anonymous> (/usr/local/lib/node_modules/azure-cli/node_modules/request/request.js:873:14)
    at Request.EventEmitter.emit (events.js:117:20)
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/azure-cli/node_modules/request/request.js:824:12)
    at IncomingMessage.EventEmitter.emit (events.js:117:20)
    at _stream_readable.js:920:16
error:   vm create-from command failed

$
```

**Comment by kmouss** [comment]
Thank you.Media link is already associated with another VM (from which you exported). Please use a different media link for the new VM creation.

**Comment by kmouss** [comment]
Closing for now. Please reactivate if necessary - see comment above - Thanks

**Comment by alexnauda** [comment]
@kmouss I experienced the same issue, and I don't understand your explanation. You say the MediaLink is associated with another VM, the one I exported. However, I deleted that VM. Is it possible that the MediaLink stays around after the VM is deleted?

I worked around this issue by removing the following text in two places in the exported json:
,"iOType":"Standard"

I believe the inclusion of that field during export is a bug in the xplat-cli.

**Comment by amarzavery** [comment]
#1635

**Comment by amarzavery** [comment]
@kmouss 
@alexnauda -  You are right. I did the same thing. Manually edited the exported json file to remove iotype references from it. When vm create-from was given the edited file, I was able to create the vm successfully. 

The bug is in the export command and it needs to be fixed. My guess, is that something changed on the server side and it does not require an explicit reference to IOType any more.
